analyzers/src/SonarAnalyzer.CSharp/Rules/CheckArgumentException.cs

```diff
@@ -153,11 +153,15 @@ static IEnumerable<string> AccessorIdentifierNames(AccessorDeclarationSyntax acc
             }
 
             static IEnumerable<string> ParentParameterList(SyntaxNode node) =>
-                node?.Ancestors().OfType<BaseTypeDeclarationSyntax>().FirstOrDefault() is { } baseTypeAncestor
-                && RecordDeclarationSyntaxWrapper.IsInstance(baseTypeAncestor)
-                && ((RecordDeclarationSyntaxWrapper)baseTypeAncestor).ParameterList is { } recordParameterList
-                    ? IdentifierNames(recordParameterList)
-                    : Enumerable.Empty<string>();
+                node?.Ancestors().OfType<BaseTypeDeclarationSyntax>().FirstOrDefault() switch
+                {
+                    var type when RecordDeclarationSyntaxWrapper.IsInstance(type) => GetIdentifierNames(((RecordDeclarationSyntaxWrapper)type).ParameterList),
+                    var type when ClassDeclarationSyntaxWrapper.IsInstance(type) => GetIdentifierNames(((ClassDeclarationSyntaxWrapper)type).ParameterList),
+                    _ => Enumerable.Empty<string>(),
+                };
```

Comment:
Here you should use the `TypeDeclarationSyntaxWrapper`. It works for records (in all compiler versions), classes, structs, and so on.
```suggestion
                node?.Ancestors().OfType<TypeDeclarationSyntax>().FirstOrDefault() is { } typeDeclaration
                    ? GetIdentifierNames(((TypeDeclarationSyntaxWrapper)typeDeclaration).ParameterList)
                    : Enumerable.Empty<string>(),
```